src/main/java/ws/wamp/jawampa/WampClient.java

```diff
@@ -1028,22 +1031,51 @@ public void call(Throwable t1) {
      * of the call be Observable&lt;String&gt;.
      * @return An observable that can be used to subscribe on the topic.
      */
-    public <T> Observable<T> makeSubscription(final String topic, final Class<T> eventClass)
+    public <T> Observable<T> makeSubscription(final String topic, final Class<T> eventClass) {
+        return makeSubscription(topic, null, eventClass);
+    }
+
+    /**
+     * Returns an observable that allows to subscribe on the given topic.<br>
+     * The actual subscription will only be made after subscribe() was called
+     * on it.<br>
+     * This version of makeSubscription will automatically transform the
+     * received events data into the type eventClass and will therefore return
+     * a mapped Observable. It will only look at and transform the first
+     * argument of the received events arguments, therefore it can only be used
+     * for events that carry either a single or no argument.<br>
+     * Received publications will be pushed to the Subscriber via it's
+     * onNext method.<br>
+     * The client can unsubscribe from the topic by calling unsubscribe() on
+     * it's Subscription.<br>
+     * If the connection closes onCompleted will be called.<br>
+     * In case of errors during subscription onError will be called.
+     * @param topic The topic to subscribe on.<br>
+     * Must be valid WAMP URI.
+     * @param options A WAMP options dictionary. May contain advanced options
+     * for the subscribe call.
+     * @param eventClass The class type into which the received event argument
+     * should be transformed. E.g. use String.class to let the client try to
+     * transform the first argument into a String and let the return value of
+     * of the call be Observable&lt;String&gt;.
+     * @return An observable that can be used to subscribe on the topic.
+     */
+    public <T> Observable<T> makeSubscription(final String topic, ObjectNode options, final Class<T> eventClass)
```

Comment:
I don't think it makes sense to expose the options field for this high level API. Because the only use-case right now is to insert a subscription pattern. However then we would possibly become different kinds of subscription data for different topics, where only one of them corresponds to `T`. And the user does not even see the topic to which the data corresponds. On the first incompatible data type the stream would close with an error.

For the lower level API which returns a `PubSubData` class with the raw information which includes the topic introducing the `options` parameter makes more sense.